File: src/com/easy/MaxConsecutiveOnes_485.java
/**
 * 
 */
package com.easy;

/**
 * @author Chris Lin
 * @version 1.0
 */
public class MaxConsecutiveOnes_485 {
    // 2019年7月23日
    // Use iterative
    // Time Complexity: O(n)
    // Space Complexity:O(1)
    // Runtime: 2 ms, faster than 99.67%
    // Memory Usage: 38.2 MB, less than 100.00%
    public int findMaxConsecutiveOnes(int[] nums) {
        int mResult = 0;
        int mTmp = 0;

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 1) {
                mTmp++;
            } else { // 0 case
                mResult = Math.max(mTmp, mResult);
                mTmp = 0;
            }
        }
        mResult = Math.max(mTmp, mResult);

        return mResult;
    }

}

File: Cpp_src/easy/max_consecutive_ones_485.cc
#include <iostream>
#include <vector>
using namespace std;

// Use loop judge:
// Time Complexity: O(n)
// Space Complexity:O(1)
// Runtime: 36 ms, faster than 84.61%
// Memory Usage: 11.8 MB, less than 72.73%
class Solution
{
public:
    int findMaxConsecutiveOnes(vector<int> &nums)
    {
        int result = 0;
        int tmp = 0;

        for (int num : nums)
        {
            if (num == 1)
            {
                tmp++;
            }
            else
            {
                result = max(result, tmp);
                tmp = 0;
            }
        }
        result = max(result, tmp);

        return result;
    }
};